### PR TITLE
OKTA-705616 Provisioning connection guide updates with OAuth2 API access

### PR DESCRIPTION
**Pull request by vanngo-okta**
## Description:
- **What's changed?** Provisioning connection guide updates with OAuth2 API access
- **Is this PR related to a Monolith release?** Yes, 2024.04.0

### Preview:
https://66032a6faa38178c2e76f57f--reverent-murdock-829d24.netlify.app/docs/guides/app-provisioning-connection/main/

### Resolves:

* [OKTA-705616](https://oktainc.atlassian.net/browse/OKTA-705616)

### CODE REVIEW

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md](https://github.com/okta/okta-developer-docs/blob/94118fa7eb1be7cd0425355baa4fe6e692a1ada8/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fapp-provisioning-connection%2Fmain%2Findex.md) | [82](https://okta.acrolinx.cloud/api/v1/checking/scorecards/1b17c141-6909-444a-a718-192febc58bf1) | 87  | 52  | 72  | 100  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.b2e2aed1-76cf-4425-8590-97a59abc3676?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md](https://github.com/okta/okta-developer-docs/blob/9cd1849fd7c72b2a5921502679e9da533e2a4aa4/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fapp-provisioning-connection%2Fmain%2Findex.md) | [84](https://okta.acrolinx.cloud/api/v1/checking/scorecards/bb4d7c2d-f94e-40eb-8873-086836aa9ed7) | 94  | 56  | 77  | 100  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.d1392c68-9d50-4be0-8969-30bca8400a5b?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md](https://github.com/okta/okta-developer-docs/blob/1ade9e2e5cf0afa58f2bbfca6965fa669657bc46/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fapp-provisioning-connection%2Fmain%2Findex.md) | [85](https://okta.acrolinx.cloud/api/v1/checking/scorecards/8885d664-5fbc-4b7a-bf5b-bf3db527bdd9) | 94  | 56  | 77  | 100  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.7589a111-1e22-4f77-8365-19dcec875f25?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md [hunk 1]
```diff
@@ -27,13 +27,45 @@ This guide shows you how to use the Okta APIs to configure and enable an Okta pr
...
+Okta supports token-based or OAuth 2.0-based authentication for the provisioning connections. The following are Okta-supported app integrations and their provisioning connection authentication types.
 
 | Connection authentication | <div style="width:300px">Apps supported</div> | Description |
 | ------------------------- | -------------- | ----------- |
 | Token | Okta Org2Org (`okta_org2org`)<br>Zscaler 2.0 (`zscalerbyz`) | The provisioning API connection is based on bearer token authentication. For example, with an API key. |
 | OAuth 2.0 | Google Workspace (`google`)<br>Microsoft Office 365 (`office365`)<br>Okta Org2Org (`okta_org2org`)<br>Slack (`slack`)<br>Zoom (`zoomus`) | The provisioning API connection is based on OAuth 2.0 authentication. |
 
+## Access Okta APIs
+
+First, you need access to the Okta APIs regardless if you're setting up a token-based or an OAuth 2.0-based provisioning connection. You can access Okta APIs with OAuth 2.0 authentication either as a service app or a user.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md:39]:
```suggestion
First, you need access to the Okta APIs regardless whether you're setting up a token-based or an OAuth 2.0-based provisioning connection. You can access Okta APIs with OAuth 2.0 authentication either as a service app or a user.
```

File: packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md [hunk 2]
```diff
@@ -27,13 +27,45 @@ This guide shows you how to use the Okta APIs to configure and enable an Okta pr
...
+
+### Get Okta API access as a service
+
+To access Okta APIs with OAuth 2.0 in a service-based context, you must create a service app in Okta. You can specify the scopes and resources granted to the service app in Okta, which allows you to obtain an access token with those grants.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md:43]:
```suggestion
To access Okta APIs with OAuth 2.0 in a service-based context, create a service app in Okta. You can specify the scopes and resources granted to the service app, which allows you to obtain an access token with those grants.
```

File: packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md [hunk 3]
```diff
@@ -27,13 +27,45 @@ This guide shows you how to use the Okta APIs to configure and enable an Okta pr
...
+* `okta.apps.read`
+* `okta.apps.manage`
+
+Ensure that you assign the **Super Administrator** role to your service app.
+
+These scopes and roles allow you to manage the app provisioning connection and features. After you obtain your access token, proceed to the [token-based](#token-based-provisioning-connection) or the [OAuth 2.0-based](#oauth-2-0-based-provisioning-connection) provisioning connection task.
+
+### Get Okta API access as a user
+
+To access Okta APIs with OAuth 2.0 in a user-based context, you must create an OIDC app in Okta for you to sign in to. You can specify the scopes and resources granted to the OIDC app in Okta, which allows you to obtain an access token with those grants.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md:58]:
```suggestion
To access Okta APIs with OAuth 2.0 in a user-based context, create an OIDC app in Okta for you to sign in to. You can specify the scopes and resources granted to the OIDC app, which allows you to obtain an access token with those grants.
```

File: packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md [hunk 4]
```diff
@@ -27,13 +27,45 @@ This guide shows you how to use the Okta APIs to configure and enable an Okta pr
...
+
+See [User-based API access setup](/docs/reference/rest/#user-based-api-access-setup) for complete steps.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md:60]:
```suggestion
See [User-based API access setup](/docs/reference/rest/#user-based-api-access-setup).
```

File: packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md [hunk 5]
```diff
@@ -229,73 +262,86 @@ After you create your app integration in Okta, configure the provisioning connec
...
-                    "status": "ENABLED"
-                },
-                "profile":{
-                    "status": "ENABLED"
+1. Configure your provisioning features. See
+ [Configure the provisioning features for your app](#configure-the-provisioning-features-for-your-app).
+
+## Configure the provisioning features for your app
+
+After you successfully activated your provisioning connection, use the [Application Features API](https://developer.okta.com/docs/api/openapi/okta-management/management/tag/ApplicationFeatures/) to configure INBOUND_PROVISIONING and/or USER_PROVISIONING features.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md:270]:
```suggestion
After you successfully activate your provisioning connection, use the [Application Features API](https://developer.okta.com/docs/api/openapi/okta-management/management/tag/ApplicationFeatures/) to configure the INBOUND_PROVISIONING and/or USER_PROVISIONING features.
```

**Review by susanharper-okta — COMMENTED** [review]
couple things

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/app-provisioning-connection/main/index.md](https://github.com/okta/okta-developer-docs/blob/36cb8ffb50cec9d4acf25c637fc98c6e3c4d8075/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fapp-provisioning-connection%2Fmain%2Findex.md) | [88](https://okta.acrolinx.cloud/api/v1/checking/scorecards/99b6cec6-77df-48d5-ab9f-ad95675908ac) | 94  | 80  | 76  | 100  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.703d5844-d95a-4d3c-b412-0b046d18d03c?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by vanngo-okta** [comment]
SME approved in Jira

**Review by susanharper-okta — APPROVED** [review]
LGTM

**Pull request merged**